initial-ai-src/script.js

```diff
@@ -0,0 +1,587 @@
+                            ? `<div style="color: #dc3545; font-size: 0.9rem;">+${Math.abs(
+                                option.savings
+                              ).toFixed(0)} NOK more</div>`
+                            : ""
+                        }
+                        ${
+                          option.type !== "reis" && option.ticketsNeeded > 1
+                            ? `<div style="color: #666; font-size: 0.8rem; margin-top: 0.5rem;">${option.ticketsNeeded} tickets needed</div>`
+                            : ""
+                        }
+                    </div>
+                `
+                  )
+                  .join("")}
+            </div>
+
+            <div style="margin-top: 1.5rem; padding: 1rem; background: #e3f2fd; border-radius: 6px;">
+                <strong>💡 Recommendation:</strong>
+                ${this.getRecommendationText(comparisons)}
+            </div>
+        `;
+  }
+
+  getZoneText(zone) {
+    const zoneMap = {
+      1: "Zone 1 (Oslo city center)",
+      2: "Zones 1-2 (Oslo + inner suburbs)",
+      3: "Zones 1-3 (Oslo + outer suburbs)",
+      4: "Zones 1-4 (Oslo + extended area)",
+      all: "All zones (Oslo + Akershus)",
+    };
+    return zoneMap[zone] || "Unknown zone";
+  }
+
+  getPeriodText(period) {
+    switch (period) {
+      case 1:
+        return "day";
+      case 7:
+        return "week";
+      case 30:
+        return "month";
+      case 365:
+        return "year";
+      default:
+        return `${period} days`;
+    }
+  }
+
+  getRecommendationText(comparisons) {
+    const recommended = comparisons.find((c) => c.isRecommended);
+    const cheapest = comparisons[0]; // Already sorted by cost
+
+    if (recommended) {
+      if (recommended.type === "reis") {
+        return "Stick with Reis single tickets - you're getting good discounts for your usage pattern!";
+      } else {
+        return `Consider buying a ${
+          recommended.name
+        } to save ${recommended.savings.toFixed(
+          0
+        )} NOK compared to single tickets.`;
+      }
+    } else if (cheapest.type === "reis") {
+      return "Reis single tickets with discounts are your best option for this usage pattern.";
+    } else {
+      return `A ${cheapest.name} would be cheapest, but the savings might not justify the upfront cost.`;
+    }
+  }
+
+  calculateBreakEvenPoint(
+    periodTicketPrice,
+    periodDays,
+    ticketPrice,
+    currentTicketCount
+  ) {
+    // Find the break-even point where Reis total cost equals period ticket cost
+    let trips = 1;
+    let totalCost = 0;
+
+    while (trips <= 200) {
+      // Reasonable upper limit
+      const breakdown = this.calculateDiscountBreakdown(
+        trips,
+        ticketPrice,
+        currentTicketCount
+      );
+      totalCost = breakdown.reduce((sum, tier) => sum + tier.totalCost, 0);
+
+      if (totalCost >= periodTicketPrice) {
+        return trips;
+      }
+      trips++;
+    }
```

Comment:
❓ Why do you have a preference for `while` loops? `for` loops for this purpose are more typical in JavaScript. But maybe there's something I'm missing?